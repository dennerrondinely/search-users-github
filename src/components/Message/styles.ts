import styled, { css } from 'styled-components';
import media from 'styled-media-query';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    align-items: center;
    gap: ${theme.spacings.xsmall};
  `}
`;

export const Title = styled.h1`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    color: ${theme.colors.secondary};
    text-align: center;

    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.xlarge};
    `}
  `}
`;

export const TextWrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    align-items: center;
    gap: ${theme.spacings.xxsmall};
    font-size: ${theme.font.sizes.xsmall};
    color: ${theme.colors['primary-light']};
    text-align: center;

    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.medium};
    `}
  `}
`;
