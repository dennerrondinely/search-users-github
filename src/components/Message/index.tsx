import React, { ReactNode } from 'react';

import * as S from './styles';

export type MessageProps = {
  title: string;
  children: ReactNode;
};

const Message = ({ title, children }: MessageProps) => {
  return (
    <S.Wrapper>
      <S.Title>{title}</S.Title>
      <S.TextWrapper>{children}</S.TextWrapper>
    </S.Wrapper>
  );
};

export default Message;
