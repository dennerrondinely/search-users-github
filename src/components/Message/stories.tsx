import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';

import Message, { MessageProps } from '.';

export default {
  title: 'Message',
  component: Message,
  argTypes: {
    children: { type: '' }
  }
} as Meta;

export const Default: Story<MessageProps> = (args) => <Message {...args} />;

Default.args = {
  title: 'Desculpe, nenhum resultado encontrado :(',
  children: (
    <>
      <p>O Usuário que você pesquisou</p>
      <p>infelizmente não foi encontrado ou não existe.</p>
    </>
  )
};
