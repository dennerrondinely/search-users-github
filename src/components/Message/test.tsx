import React from 'react';
import { render, screen } from 'utils/tests/helpers';

import Message from './index';

describe('<Message />', () => {
  it('should render the Message', () => {
    const { container } = render(
      <Message title="Desculpe, nenhum resultado encontrado :(">
        <p role="paragraph">O Usuário não foi encontrado</p>
      </Message>
    );

    expect(screen.getByRole('heading')).toHaveTextContent(
      'Desculpe, nenhum resultado encontrado :('
    );

    expect(screen.getByRole('paragraph')).toHaveTextContent(
      'O Usuário não foi encontrado'
    );

    expect(container).toContainHTML('p');

    expect(container).toMatchSnapshot();
  });
});
