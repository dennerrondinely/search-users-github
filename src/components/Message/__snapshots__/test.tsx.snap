// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> should render the Message 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1.6rem;
}

.c1 {
  font-size: 1.6rem;
  color: #2B716E;
  text-align: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.8rem;
  font-size: 1.2rem;
  color: #7c7570;
  text-align: center;
}

@media (min-width:768px) {
  .c1 {
    font-size: 2.0rem;
  }
}

@media (min-width:768px) {
  .c2 {
    font-size: 1.6rem;
  }
}

<div>
  <div
    class="c0"
  >
    <h1
      class="c1"
    >
      Desculpe, nenhum resultado encontrado :(
    </h1>
    <div
      class="c2"
    >
      <p
        role="paragraph"
      >
        O Usuário não foi encontrado
      </p>
    </div>
  </div>
</div>
`;
